llama_stack/distribution/providers.py

```diff
@@ -57,3 +66,63 @@ async def inspect_provider(self, provider_id: str) -> ProviderInfo:
                 return p
 
         raise ValueError(f"Provider {provider_id} not found")
+
+    async def get_providers_health(self) -> Dict[str, Dict[str, HealthResponse]]:
+        """Get health status for all providers.
+
+        Returns:
+            Dict[str, Dict[str, HealthResponse]]: A dictionary mapping API names to provider health statuses.
+                Each API maps to a dictionary of provider IDs to their health responses.
+        """
+        providers_health: Dict[str, Dict[str, HealthResponse]] = {}
+        timeout = 1.0
+
+        async def check_provider_health(impl: Any) -> tuple[str, HealthResponse]:
+            """Check health of a single provider.
+
+            Args:
+                impl: Provider implementation instance
+
+            Returns:
+                tuple[str, HealthResponse]: (api_name, health_response)
+            """
+            api_name = impl.__provider_spec__.api.name
+
+            if not hasattr(impl, "health"):
+                return (
+                    api_name,
+                    HealthResponse(
+                        status=HealthStatus.NOT_IMPLEMENTED, message="Provider does not implement health check"
+                    ),
+                )
+
+            try:
+                await impl.initialize()
```

Comment:
this seems not quite right. the lifecycle agreement (never documented any place ever 😭) is that the server (resolver) instantiates the providers and initializes them _once_. nothing outside that code should do anything with the lifecycle. it is certainly not expected that `initialize()` should be idempotent, etc. 

what happens if you don't call `initialize()` anywhere and depend on the server having done the right thing?

Reply:
It's working without this line, thanks!